### ISSUE TITLE
Elastic training supports AllReduce strategy

### ISSUE THREAD

**Issue by yifeng-x** [intelligent-machine-learning/dlrover#1573]
Requirements:
To implement distributed elastic training capabilities supporting the AllReduce strategy (synchronous) in PyTorch or PyTorch+DeepSpeed frameworks.

Application Scenarios: 
To be applied in large model training scenarios, utilizing automatic elastic scaling capabilities to dynamically adjust resource sizes, accelerate model training, and improve resource utilization.

**Comment by yifeng-x** [comment]
I understand that DLRover currently only supports fault tolerance in AllReduce scenarios, and does not support elastic scaling of resources.

**Comment by BalaBalaYi** [comment]
> I understand that DLRover currently only supports fault tolerance in AllReduce scenarios, and does not support elastic scaling of resources.

Torch's MPI-style distributed computing flexibility is related to many factors, including GPU, NCCL, and Torch's implementation. Therefore, DLRover definitely cannot support this at the current stage (this is not something DLRover can solve).  TensorFlow's support for this is because its training workers can be considered stateless.

**Comment by workingloong** [comment]
Now, DLRover can automatically scale up/down a PyTorch allreduce job according to the avaliable resource in the cluster. Howerver, in an synchronous allreduce job, the global batch size is equal to the local world size * the number of nodes. If the number of nodes changes, the global batch size changes with the number of nodes. The global batch size can affect the model precision. This is the main challenge to dynamically adjust the resource for an synchronous allreduce job. Though, we can adjust the local batch size by accumulating gradient, but it may hurt the training efficiency.

**Comment by KuaiRuijie** [comment]
> Now, DLRover can automatically scale up/down a PyTorch allreduce job according to the avaliable resource in the cluster. Howerver, in an synchronous allreduce job, the global batch size is equal to the local world size * the number of nodes. If the number of nodes changes, the global batch size changes with the number of nodes. The global batch size can affect the model precision. This is the main challenge to dynamically adjust the resource for an synchronous allreduce job. Though, we can adjust the local batch size by accumulating gradient, but it may hurt the training efficiency.

If I understand correctly, you're suggesting that when handling elastic scaling for synchronous AllReduce training, our priority should be to maintain the global batch size to preserve model precision, and then address the impact on training efficiency caused by adjusting the local batch size. Is that right?
